helpick-service/src/test/java/com/infoshare/service/VolunteerServiceTest.java

```diff
@@ -0,0 +1,29 @@
+package com.infoshare.service;
+
+import com.infoshare.database.DB;
+import com.infoshare.database.FileDb;
+import com.infoshare.domain.TypeOfHelp;
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+
+
+class VolunteerServiceTest {
+
+    private final DB db = new FileDb();
+    private final VolunteerService volunteerService = new VolunteerService(db);
+
+    @Test
+    public void ifNewVolunteerRegisteredForValidInput() {
+        assertTrue(volunteerService.registerNewVolunteer("Janko", "Gdansk", "janko@gmail.com", "123654789", TypeOfHelp.SHOPPING, true));
+        db.deleteVolunteer("janko@gmail.com");
+    }
+
+    @Test
+    public void ifNewVolunteerRegisteredWithEmailAlreadyUsedInDb(){
+        assertTrue(volunteerService.registerNewVolunteer("Janko", "Gdansk", "janko@gmail.com", "123654789", TypeOfHelp.SHOPPING, true));
+        assertFalse(volunteerService.registerNewVolunteer("Anna", "Warszawa", "janko@gmail.com", "123654789", TypeOfHelp.SHOPPING, true));
+        db.deleteVolunteer("janko@gmail.com");
```

Comment:
Jeśli test ma taką strukturę, to ciężko zrozumieć co chcemy tak naprawdę przetestować. Lepiej to poukładać np. w ten sposób:
```
@Test
public void ifNewVolunteerRegisteredWithEmailAlreadyUsedInDb(){
    // Given
    var volunteer = new Volunteer("Janko", "Gdansk", "janko@gmail.com", "123654789", TypeOfHelp.SHOPPING, true, UUID.randomUUID());
    db.saveVolunteer(volunteer);

    // When
    boolean isVolunteerRegistered = volunteerService.registerNewVolunteer("Anna", "Warszawa", "janko@gmail.com", "123654789", TypeOfHelp.SHOPPING, true);

    // Then
    assertThat(isVolunteerRegistered).isFalse();
}
```

Oczywiście jeszcze lepiej byłoby, gdybyśmy używali mocka DB zamiast faktycznej bazy plikowej, bo wówczas nie musimy zapisywać wolontariusza do bazy w sekcji given, tylko ustawilibyśmy dla tego mocka odpowiednie zachowanie:
```
@Test
public void ifNewVolunteerRegisteredWithEmailAlreadyUsedInDb(){
    // Given
    when(db.getVolunteer("janko@gmail.com")).thenReturn(new Volunteer("Janko", "Gdansk", "janko@gmail.com", "123654789", TypeOfHelp.SHOPPING, true, UUID.randomUUID()));

    // When
    boolean isVolunteerRegistered = volunteerService.registerNewVolunteer("Anna", "Warszawa", "janko@gmail.com", "123654789", TypeOfHelp.SHOPPING, true);

    // Then
    assertThat(isVolunteerRegistered).isFalse();
}
```
Test z mockiem zadziała tylko wtedy, gdy obiekt db zadeklarujemy jako mock:
`private final DB db = mock(DB.class);`